build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/CycleSchemaGenerator.java

```diff
@@ -235,12 +235,7 @@ protected String getSchemaClassTemplateName() {
     public List<AttributeType> sort() {
         List<AttributeType> sorted = new ArrayList<>(
                 types.values());
-        Collections.sort(sorted, new Comparator<AttributeType>() {
-            @Override
-            public int compare(AttributeType at1, AttributeType at2) {
-                return ((NameImpl) at1.getName()).compareTo((NameImpl) at2.getName());
-            }
-        });
+        Collections.sort(sorted, (at1, at2) -> ((NameImpl) at1.getName()).compareTo((NameImpl) at2.getName()));
```

Comment:
This is more what I was expecting, is the cast to NameImpl needed?

Reply:
No idea, the xmlcodegen modules has been written by someone else... that said, the cast to NameImpl was already in the orginal code. I'll notice NameImpl is not GeoTools one, it does not even have the "getName" method.